Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm

```diff
@@ -4716,19 +4723,20 @@ static bool selectionIsInsideFixedPositionContainer(LocalFrame& frame)
         if (largeViewportConstrainedElements.isEmpty())
             return false;
 
-        RefPtr hitTestedNode = mainFrame->eventHandler().hitTestResultAtPoint(LayoutPoint { viewportRect.center() }, HitTestRequest::Type::ReadOnly).innerNode();
-        if (!hitTestedNode)
+        auto hitTestResult = HitTestResult { LayoutRect { viewportRect } };
+        if (!document->hitTest({ HitTestSource::User, { HitTestRequest::Type::ReadOnly, HitTestRequest::Type::CollectMultipleElements } }, hitTestResult))
             return false;
 
-        return largeViewportConstrainedElements.containsIf([hitTestedNode](auto& element) {
-            return element->contains(*hitTestedNode);
-        });
+        for (auto& element : largeViewportConstrainedElements) {
+            if (hitTestResult.listBasedTestResult().contains(element))
+                return true;
+        }
+
+        return false;
```

Comment:
Totally optional, but this portion is a nice candidate to write with `std::ranges::any_of`.

```cpp
return std::ranges::any_of(largeViewportConstrainedElements, [listBasedTestResult = hitTestResult.listBasedTestResult()](auto& element) {
    return listBasedTestResult.contains(element);
});
```